tests/test_formatted_json.py

```diff
@@ -0,0 +1,305 @@
+from argparse import Namespace
+
+import pytest
+from flake8.violation import Violation
+
+from flake8_json_reporter.reporters import FormattedJSON
+
+
+@pytest.fixture
+def formatter():
+    """Return a ``FormattedJSON`` instance."""
+    options = Namespace(output_file=None, color=False, tee=False)
+    formatter = FormattedJSON(options)
+    return formatter
+
+
+@pytest.fixture
+def dummy_violation():
+    return Violation(
+        code="E222",
+        filename="main.py",
+        line_number=42,
+        column_number=4,
+        text="multiple spaces after operator",
+        physical_line="x =  1",
+    )
+
+
+def run(formatter, violations):
+    formatter.start()
+    for filename in violations:
+        formatter.beginning(filename)
+        for violation in violations[filename]:
+            formatter.format(violation)
+        formatter.finished(filename)
+    formatter.stop()
+
+
+def test_no_files(capsys, formatter):
+    run(formatter, {})
+    stdout, _ = capsys.readouterr()
+    assert stdout == "{}\n"
+
+
+def test_single_file_no_violations(capsys, formatter):
+    run(formatter, {"main.py": []})
+    stdout, _ = capsys.readouterr()
+    assert (
+        stdout
+        == """\
+{
+  "main.py": []
+}
+"""
+    )
+
+
+def test_multiple_files_no_violations(capsys, formatter):
+    run(formatter, {"main.py": [], "__init__.py": []})
+    stdout, _ = capsys.readouterr()
+    assert (
+        stdout
+        == """\
+{
+  "main.py": [],
+  "__init__.py": []
+}
+"""
+    )
+
+
+def test_single_file_single_violation(capsys, formatter, dummy_violation):
+    run(
+        formatter,
+        {
+            "main.py": [dummy_violation],
+        },
+    )
+    stdout, _ = capsys.readouterr()
+    assert (
+        stdout
+        == """\
+{
+  "main.py": [
+    {
+      "code": "E222",
+      "filename": "main.py",
+      "line_number": 42,
+      "column_number": 4,
+      "text": "multiple spaces after operator",
+      "physical_line": "x =  1"
+    }
+  ]
+}
+"""
+    )
```

Comment:
I would write these as:
```
    expected = """\
...
...
"""
    assert out == expected
```

that'll be a little less weird for black

Reply:
Thanks for the review! I applied your suggestions :)